docker/testing/ubuntu2204_java_driver_testing

```diff
@@ -0,0 +1,62 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM ubuntu:22.04
+MAINTAINER Apache Cassandra <dev@cassandra.apache.org>
+
+RUN export DEBIAN_FRONTEND=noninteractive && \
+	apt update  && \
+	apt install -y curl git virtualenv software-properties-common vim maven sudo && \
+	add-apt-repository -y ppa:deadsnakes/ppa && \
+	apt update
+
+# set up python 3.9
+RUN DEBIAN_FRONTEND=nointeractive apt install -y python3.9
+RUN update-alternatives --install /usr/bin/python python /usr/bin/python3.9 1
+
+# add user
+RUN useradd -rm -d /home/docker -s /bin/bash -g root -G sudo -u 1000 -p "$(openssl passwd -1 docker)" docker
+RUN echo "docker ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+
+USER docker
+
+WORKDIR /home/docker
+
+# install jabba and java versions
+RUN export JABBA_HOME="$HOME/.jabba" && \
+	curl -sL https://github.com/shyiko/jabba/raw/master/install.sh | bash && . "$HOME/.jabba/jabba.sh" && \
+	jabba install zulu@1.8 && \
+	jabba install openjdk@1.11.0 && \
+	jabba install openjdk@1.17.0
+
+ENV PATH="/home/docker/.jabba/bin:/home/docker/.local/bin:$PATH"
+
+RUN echo "export JAVA8_HOME=$(jabba which zulu@1.8)" >> ~/env.txt && \
+  	echo "export JAVA11_HOME=$(jabba which openjdk@1.11.0)" >> ~/env.txt && \
+	  echo "export JAVA17_HOME=$(jabba which openjdk@1.17.0)" >> ~/env.txt && \
+  	echo "export JAVA_HOME=$(jabba which zulu@1.8)" >> ~/env.txt && \
+  	echo ". $HOME/.jabba/jabba.sh" >> ~/env.txt && \
+  	echo "jabba use zulu@1.8" >> ~/env.txt
+
+# ccm
+RUN git clone https://github.com/riptano/ccm.git  && \
+	cd ccm && \
+	pip install -e .
+
+# driver
+RUN git clone https://github.com/apache/cassandra-java-driver.git
```

Comment:
how will you switch to the git fork and branch that you want to be testing …?

Reply:
When used as a user's development environment, I thought the user can just add their own git remote and switch to the branch for development. As used in the Jenkins job, previously it worked for the jenkins job because jenkins checkout the fork/branch for me when I used the docker image at the agent level.
Mounting the current directory also sounds good to me. Anything I need to change here for the docker definition?

Reply:
see my branch of your cassandra-java-driver patch.
i mount the current directory into docker
https://github.com/SiyaoIsHiding/java-driver/compare/asf-ci-3.x...thelastpickle:cassandra-java-driver:mck/Siyaso/asf-ci-3.x#diff-6e6a535c03e06e3c8fbb6c401d7daac3eb1df13f4664267168521f31f1cc3df8R24

with that in place you don't need line 56 above (you don't need to git clone inside the image)*

ref: https://github.com/SiyaoIsHiding/java-driver/compare/asf-ci-3.x...thelastpickle:cassandra-java-driver:mck/Siyaso/asf-ci-3.x

*) a valid reason to git clone is if you want to do a build so that ~/.m2/repository has most of the dependencies ready to go in the image.